src/test/java/com/android/automation/training/steps/SearchSteps.java

```diff
@@ -55,149 +37,28 @@ public void userClickTheProductSuggestionBelowTheSearchBar() {
     @And("user search the product {string} on the search bar")
     public void userSearchTheProductOnTheSearchBar(String input_search) {
         By INPUT_KEYWORD = MobileBy.id("edt_search");
-        input(INPUT_KEYWORD,input_search);
+        input(INPUT_KEYWORD, input_search);
     }
 
     @And("user see the product suggestion {string} below the search bar")
     public void userSeeTheProductSuggestionBelowTheSearchBar(String see_product) {
         By SUGGESTION = MobileBy.id("tv_keyword");
-        Assertions.assertEquals(see_product,getText(SUGGESTION),"Failed load suggestion");
+        Assertions.assertEquals(see_product, getText(SUGGESTION), "Failed load suggestion");
     }
 
     @Then("user should discover the product {string} on the search result page")
     public void userShouldDiscoverTheProductOnTheSearchResultPage(String title_product) {
 
-     By RESULT = MobileBy.id("txt_product_name");
+        By RESULT = MobileBy.id("txt_product_name");
         findAll(RESULT);
```

Comment:
create new variable to store arraylist data 
List<String>actualData = findAll().stream.map(RemoteWebElement::getText()) .collect(Collectors.toList());